test_programs/execution_success/brillig_loop_size_regression/src/main.nr

```diff
@@ -7,10 +7,8 @@ struct EnumEmulation {
 unconstrained fn main() -> pub Field {
     let mut emulated_enum = EnumEmulation { a: Option::some(1), b: Option::none(), c: Option::none() };
 
-    // Do a copy to optimize out loads in the loop
-    let copy_enum = emulated_enum;
```

Comment:
After this PR:
Even after removing this copy optimization we get a Brillig bytecode size improvement from 55 -> 40.

Keeping the copy optimizations we get the following final SSA which produces 24 brillig opcodes:
```
After Array Set Optimizations:
brillig fn main f0 {
  b0():
    jmp b1(u32 0)
  b1(v6: u32):
    v27 = eq v6, u32 0
    jmpif v27 then: b2, else: b3
  b2():
    v28 = add v6, u32 1
    jmp b1(v28)
  b3():
    return Field 2
}
```

Reply:
Ok this PR now produces the optimal SSA specified above without the copy optimization. Just getting a failure on `uhashmap` that is the same as https://github.com/noir-lang/noir/issues/5897.